src/tox/session/env_select.py

```diff
@@ -389,6 +391,16 @@ def _mark_provision(self, on: bool, provision_tox_env: str) -> None:  # noqa: FB
         self._provision = on, provision_tox_env
 
 
+def _find_env_match(value: str, state_conf: set[str]) -> bool:
+    return any(value in conf.split("-") for conf in state_conf) or any(
+        value == "-".join(combo) for combo in permutations(state_conf, 2)
```

Comment:
Would this work for permutations of 3? Should we just check that any factor matches the regex:

```
(py)|(\d+([.]\d+([.]\d+)?)?)|(py\d+([.]\d+([.]\d+)?)?)
```

I feel would generalize better 🤔

Reply:
What's the maximum number of factors? Given that you don't create the "manually" by specifying an env in `env_list` that has a hyphen in it?

Reply:
I'm sorry. I'm really struggling to understand the heuristics at play here.

Reply:
There is no limit.